### PR TITLE
Improve container-based Instance controller tests

### PR DESCRIPTION
**Pull request by lucafrancescato**
# Description

This PR aims to improve the tests on the container-based Instance controller, in order to provide a guideline when further developing the operator for the container-based Instance.

### CODE REVIEW

File: operators/pkg/instance-controller/container_controller_test.go [hunk 1]
```diff
@@ -270,22 +347,26 @@ var _ = Describe("Instance Operator controller for containers", func() {
...
-				if err != nil && errors.IsNotFound(err) {
-					return true
-				}
-				return false
-			}, timeout, interval).Should(BeTrue())
+			By("Checking that the container Instance is deleted")
+			IsEventuallyNotFound(ctx, types.NamespacedName{
+				Name:      InstanceName,
+				Namespace: InstanceNamespace,
+			}, &instance, BeTrue(), timeout, interval)
```
Comment (giorio94) [inline operators/pkg/instance-controller/container_controller_test.go:359]: This part could be dropped, since it does not test the controller behavior

**Review by giorio94 — APPROVED** [review]
/lgtm

Just a minor comment inline. Before merging, please quash the commits

**Review by QcFe — APPROVED** [review]
lgtm

**Pull request merged**